msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/mwpj/TestCase2521946.kt

```diff
@@ -0,0 +1,143 @@
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.client.msal.automationapp.testpass.broker.mwpj
+
+import com.microsoft.identity.client.msal.automationapp.R
+import com.microsoft.identity.client.msal.automationapp.testpass.broker.AbstractMsalBrokerTest
+import com.microsoft.identity.client.ui.automation.annotations.LocalBrokerHostDebugUiTest
+import com.microsoft.identity.client.ui.automation.annotations.SupportedBrokers
+import com.microsoft.identity.client.ui.automation.broker.BrokerHost
+import com.microsoft.identity.client.ui.automation.interaction.PromptHandlerParameters
+import com.microsoft.identity.client.ui.automation.interaction.PromptParameter
+import com.microsoft.identity.client.ui.automation.rules.LoadLabUserTestRule
+import com.microsoft.identity.labapi.utilities.client.ILabAccount
+import com.microsoft.identity.labapi.utilities.client.LabQuery
+import com.microsoft.identity.labapi.utilities.constants.AzureEnvironment
+import com.microsoft.identity.labapi.utilities.constants.TempUserType
+import com.microsoft.identity.labapi.utilities.constants.UserType
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TestRule
+
+// https://identitydivision.visualstudio.com/Engineering/_workitems/edit/2521946
+@SupportedBrokers(brokers = [BrokerHost::class])
+@LocalBrokerHostDebugUiTest
+class TestCase2521946 : AbstractMsalBrokerTest() {
+
+    private lateinit var mUsGovLabAccount: ILabAccount
+
+    @get:Rule
+    val loadAdditionalLabUserRule: TestRule = LoadLabUserTestRule(getAdditionalLabQuery())
+
+    @Test
+    fun test_2521946() {
+
+        val username = mLabAccount.username
+        val password = mLabAccount.password
+        val usGovUsername = mUsGovLabAccount.username
+        val usGovPassword = mUsGovLabAccount.password
+        val brokerHostApp = broker as BrokerHost
+        brokerHostApp.enableMultipleWpj()
+        // Register 2 accounts from different tenants
+        brokerHostApp.performDeviceRegistrationMultiple(usGovUsername, usGovPassword)
+        brokerHostApp.performDeviceRegistrationMultiple(username, password)
+        val deviceRegistrationRecords = brokerHostApp.allRecords
+        Assert.assertEquals(2, deviceRegistrationRecords.size)
+        // Unregister the device from the legacy space
+        brokerHostApp.unregisterDeviceMultiple(usGovUsername)
+        // Verify that the device is unregistered for the first account using the legacy API
+        val errorMessage = brokerHostApp.accountUpn
+        Assert.assertNotNull(errorMessage)
+        Assert.assertTrue(errorMessage!!.contains("Device is not Workplace Joined"))
+        val recordInExtendedSpace = brokerHostApp.getRecordByUpn(username)
+        // re-register the device for the first account using the legacy API
+        val promptHandlerParameters = PromptHandlerParameters.builder()
```

Comment:
Seems like this automation is deviating from the ado case in some steps. Ado test case doesn't mention unregistering or re-registering for the first account

Reply:
Wording in this test is wrong, fix it
79ed44ed0cb3060fe9ad4fcecdfb5c289d55e546